utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.decorateWithUserProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.extractCurrentUserProfileId
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [CurrentUserProfileIdIntentDecorator]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class CurrentUserProfileIdIntentDecoratorTest {
+
+  @Test
+  fun testDecorator_decorateWithProfileId_returnsIntentWithCorrectProfileId() {
+    val intent = Intent().apply {
+      decorateWithUserProfileId(
+        ProfileId.newBuilder().apply {
+          internalId = 0
+        }.build()
+      )
+    }
+    val currentProfileId = intent.extractCurrentUserProfileId()
+    assertThat(currentProfileId.internalId).isEqualTo(0)
+  }
+
+  @Test
+  fun testDecorator_withoutProfileId_returnsIntentWithDefaultProfileId() {
+    val currentProfileId = Intent().extractCurrentUserProfileId()
+    assertThat(currentProfileId).isEqualTo(ProfileId.getDefaultInstance())
```

Comment:
Prefer using TruthProtoSubject here & elsewhere in this test suite, too, per the earlier comment.

Reply:
Updated..